custom_components/wyzeapi/cover.py

```diff
@@ -0,0 +1,188 @@
+
+
+from .const import CONF_CLIENT, DOMAIN, COVER_UPDATED
+from .token_manager import token_exception_handler
+
+_LOGGER = logging.getLogger(__name__)
+ATTRIBUTION = "Data provided by Wyze"
+SCAN_INTERVAL = timedelta(seconds=10)
+MAX_OUT_OF_SYNC_COUNT = 5
+
+
+@token_exception_handler
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry,
+                            async_add_entities: Callable[[List[Any], bool], None]) -> None:
+    """
+    This function sets up the config_entry
+
+    :param hass: Home Assistant instance
+    :param config_entry: The current config_entry
+    :param async_add_entities: This function adds entities to the config_entry
+    :return:
+    """
+
+    _LOGGER.debug("""Creating new WyzeApi cover component""")
+    client: Wyzeapy = hass.data[DOMAIN][config_entry.entry_id][CONF_CLIENT]
+    camera_service = await client.camera_service
+    cameras: List[Camera] = await camera_service.get_cameras()
+    garages = []
+    for camera in cameras:
+        if camera.device_params["dongle_product_model"] == "HL_CGDC":
+            garages.append(WyzeGarageDoor(camera_service, camera))
+
+
+    async_add_entities(garages, True)
+
+
+class WyzeGarageDoor(homeassistant.components.cover.CoverEntity, ABC):
+    """Representation of a Wyze Garage Door."""
+    _attr_device_class = CoverDeviceClass.GARAGE
+    _attr_supported_features = (
+        CoverEntityFeature.OPEN,
+        CoverEntityFeature.CLOSE
+    )
+    _attr_has_entity_name = True
+
+
+    def __init__(self, camera_service: CameraService, camera: Camera):
+        """Initialize a Wyze garage door."""
+        self._camera = camera
+        if self._camera.type not in [
+            DeviceTypes.CAMERA
+        ]:
+            raise HomeAssistantError(f"Invalid device type: {self._camera.type}")
+        
+        self._camera_service = camera_service
+        self._available = self._camera.available
+        self._out_of_sync_count = 0
+        
+    @property
+    def device_info(self):
+        """Return device information about this entity."""
+        return {
+            "identifiers": {(DOMAIN, self._camera.mac)},
+            "name": f"{self._camera.nickname}.cgdc",
+            "connections": {
+                (
+                    dr.CONNECTION_NETWORK_MAC,
+                    self._camera.mac
+                )
+            }
+        }
+    @property
+    def extra_state_attributes(self):
+        """Return device attributes of the entity."""
+        return {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            "device model": f"{self._camera.product_model}.{self._camera.device_params['dongle_product_model']}"
+        }
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        # always true because garage controllers are always connected to the cam
+        return True
+    
+    @property
+    def should_poll(self) -> bool:
+        return False
```

Comment:
Away from a computer - this needs to be true :)
I was testing some things and never set this back.

Reply:
Actually it should be false. The update manager handles this, we don't technically poll in the way HA considers it.